src/functions/listFunctions.ts

```diff
@@ -0,0 +1,60 @@
+import Table = require("cli-table");
+import * as backend from "../deploy/functions/backend";
+import { logger } from "../logger";
+import { previews } from "../previews";
+import * as args from "../deploy/functions/args";
+
+/**
+ * Lists all functions in the Firebase project
+ * @param context the Context of the project
+ * @returns mapping that contains a list of functions under the 'functions' key
+ */
+export async function listFunctions(
+  context: args.Context
+): Promise<{ functions: backend.FunctionSpec[] }> {
+  const bkend = await backend.existingBackend(context, true);
+  const functionSpecs = previews.functionsv2
+    ? bkend.cloudFunctions
+    : bkend.cloudFunctions.filter((fn) => fn.platform === "gcfv1");
+  functionSpecs.sort((fn1, fn2) => {
```

Comment:
Let's improve consistency by promoting [this method](https://github.com/firebase/firebase-tools/blob/master/src/deploy/functions/prompts.ts#L16) to something in backend.ts and use it here.

Totally unrelated to this change, I'd also like to consider changing the sorting method to sort by region tier before region name.

Reply:
I exported the compareFunctions function and used it in backend.ts where we list all the functions and I removed this file.